python/ray/data/dataset.py

```diff
@@ -42,7 +42,7 @@
 from ray.data.impl.sort import sort_impl
 from ray.data.impl.block_list import BlockList
 from ray.data.impl.lazy_block_list import LazyBlockList
-from ray.data.impl.arrow_block import DelegatingArrowBlockBuilder
+from ray.data.impl.table_block import DelegatingBlockBuilder, TableRow
```

Comment:
I think we also need to change `map_batches` to not auto convert pandas => Arrow:
```
                elif isinstance(applied, pd.core.frame.DataFrame):
                    applied = pa.Table.from_pandas(applied)
```

Reply:
We should make sure in general that data stays in the same format if possible (for transform ops, shuffle ops, etc.)

Reply:
Done in another PR.